### PR TITLE
[FLOC-2480] Add new-style doc versioning.

### PR DESCRIPTION
**Pull request by tomprince**
https://clusterhq.atlassian.net/browse/FLOC-2480

[Review on Reviewable](https://reviewable.io/reviews/clusterhq/build.clusterhq.com/107)

### CODE REVIEW

File: flocker_bb/steps.py [hunk 1]
```diff
@@ -207,7 +207,15 @@ def _isMaster(branch):
...
     _RELEASE_TAG_RE = re.compile(
-        '^[0-9]+\.[0-9]+\.[0-9]+(?:dev[0-9]+|pre[0-9]+|\+doc[0-9]+)?$')
+        # <major>.<minor>.<micro>
+        r'^[0-9]+\.[0-9]+\.[0-9]+'
+        # weekly release
+        r'(?:dev[0-9]+'
+        # prerelease
+        r'|pre[0-9]+'
+        # documentation release
+        r'|(\+doc|\.post)[0-9]+)?'
```
Comment (tomprince) [inline flocker_bb/steps.py]: I leave the old style, to allow this to be merged before ClusterHQ/flocker#1699

Comment (jongiddy) [inline flocker_bb/steps.py]: Outer parentheses are non-capturing, inner parentheses are capturing. It probably doesn't matter greatly, but we may as well be consistent.

File: flocker_bb/steps.py [hunk 2]
```diff
@@ -207,7 +207,15 @@ def _isMaster(branch):
         return branch == 'master'
 
     _RELEASE_TAG_RE = re.compile(
-        '^[0-9]+\.[0-9]+\.[0-9]+(?:dev[0-9]+|pre[0-9]+|\+doc[0-9]+)?$')
+        # <major>.<minor>.<micro>
+        r'^[0-9]+\.[0-9]+\.[0-9]+'
```
Comment (jongiddy) [inline flocker_bb/steps.py]: :question: Add
```
# plus (optionally) any of:
r'(?:'
# weekly release
r'dev[0-9]+'
```

File: flocker_bb/steps.py (hunk 1 above)
Comment (jongiddy) [inline flocker_bb/steps.py]: Outer parentheses are non-capturing, inner parentheses are capturing. It probably doesn't matter greatly, but we may as well be consistent.

**Comment by jongiddy** [comment]
This PR should include some tests, either for the `isReleaseBranch` function or for the `mergeTarget` method, demonstrating the result for each type of release version, and for one non-release version.

**Pull request merged**